internal/content_coding_test.go

```diff
@@ -113,3 +156,168 @@ func TestStreamGzipDecode(t *testing.T) {
 
 	require.Equal(t, []byte("howdy"), b[:n])
 }
+
+func BenchmarkZstdEncodeDecode(b *testing.B) {
+	enc, err := NewZstdEncoder(1)
+	require.NoError(b, err)
+	dec, err := NewZstdDecoder()
+	require.NoError(b, err)
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+	}
+}
+
+func BenchmarkZstdReuse(b *testing.B) {
+	enc, err := NewZstdEncoder(1)
+	require.NoError(b, err)
+	dec, err := NewZstdDecoder()
+	require.NoError(b, err)
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+
+		payload, err = enc.Encode([]byte("doody"))
+		require.NoError(b, err)
+
+		actual, err = dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "doody", string(actual))
```

Comment:
Just do the compression/decompression once and add a warmup before the benchmark to have the internal buffer allocated...
```suggestion
	data := []byte("This is a test message for compression and decompression benchmarking!")

	// Warm-up the decompression to not measure internal buffer allocations
	payload, err := enc.Encode(data)
	require.NoError(b, err)
	actual, err := dec.Decode(payload, maxDecompressionSize)
	require.NoError(b, err)
	require.Equal(b, data, actual)
	
	for n := 0; n < b.N; n++ {
		p, _ := enc.Encode(data)
		_, _ = dec.Decode(p, maxDecompressionSize)
```
You can do the same for the other algorithms below.

Reply:
If you want to, you can split the benchmark into one for compression and one for decompression, but for me having both in one is also ok.